### PR TITLE
Add TypeORM config

### PR DESCRIPTION
**Pull request by dortegan**

### CODE REVIEW

File: generators/typeorm/templates/ormconfig.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+const path = require('path');
+const orm = require('config').orm;
+const { username, password, database, host, dialect, port, type } = orm;
+const url = `${dialect}://${username}:${password}@${host}:${port}/${database}`;
+module.exports = {
+  url,
+  type,
+  synchronize: true,
+  entities: [
+    `${__dirname}/src/entities/{*.ts,**/!(__tests__)/*.ts}`
```
Comment (warseph) [inline generators/typeorm/templates/ormconfig.js]: can we use the `.model` pattern in the name? or does it create issues with the generators?

Comment (dortegan) [reply generators/typeorm/templates/ormconfig.js]: An entity is more than a model, what we win adding the `.something` pattern in name?. Maybe use `.entity`? @warseph

File: generators/typeorm/index.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+const Generator = require('yeoman-generator');
+const mkdirp = require('mkdirp');
+const { merge } = require('lodash');
+const yamlHelper = require('../../helpers/yaml-helper');
+const fileHelper = require('../../helpers/file-helper');
+const addScript = require('../../helpers/add-script');
+
+module.exports = class extends Generator {
+  static get dependencies() {
+    return ['typeorm', 'reflect-metadata', '@types/node', 'pg'];
```
[the remark this answers is not in the document]
Comment (dortegan) [reply generators/typeorm/index.js]: 👍

File: generators/mocha/templates/typeorm.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { getConnection } from 'typeorm';
+import { connection } from '../initialization/typeorm';
+
+const clearDb = async () => {
+  try{
+    const conn = getConnection();
+    const entities = conn.entityMetadatas;
+    for(const entity of entities) {
+      const repository = await conn.getRepository(entity.name);
+      await repository.query(`TRUNCATE TABLE ${entity.tableName };`);
```
[the remark this answers is not in the document]
Comment (dortegan) [reply generators/mocha/templates/typeorm.ts]: 👍

File: generators/typeorm/templates/ormconfig.js (hunk 1 above)
Comment (warseph) [reply generators/typeorm/templates/ormconfig.js]: mmm, no, it's ok, lets make it so this folder is *only* database entities.

Comment (dortegan) [reply generators/typeorm/templates/ormconfig.js]: 👍

**Pull request merged**